### DISCUSSION TITLE
What determines the count value of instance_jvm_cpu?

### DISCUSSION THREAD

**Discussion by BFergerson** [apache/skywalking#9572] [Q&A] [answered]
I'm trying to get a better understanding of the metric system within SkyWalking and I'm having a hard time understanding why the count values in instance_jvm_cpu acquire the values they have. Here are some examples:
![image](https://user-images.githubusercontent.com/3278877/188635853-6586b435-eb6c-446e-8e74-183b47abfd79.png)
> Ignore the top two values, I'm interested in the per minute counts.

From my understanding of the code, I would expect the count values to reach a maximum of `60`. I believe this because `JVMService` publishes updates once per second. I only have one JVM running so I would expect that the count increments predictably at one per second. Instead, the pattern is much less predictable.

So my question is, what exactly determines what the `count` value of `instance_jvm_cpu`? Thanks.

**Comment by wu-sheng** [comment, marked as answer]
The value is calculated in the agent .

Why the value should increase? The thing is avg for all reported data in the time slot.

**Reply by BFergerson** [reply]
Are you referring to the `count` when you say value? Because I do not believe `count` is calculated by the agent.

I understand why `double_value` is what it is. What I'm asking is how `count` comes to be something like `1707`. The only place I see `count` incremented is `InstanceJvmCpuMetrics.combine()` and the param `count` is even annotated with `@ConstOne`.

Doesn't that mean `InstanceJvmCpuMetrics.combine()` was called `1707` times? My question is why would it be called that frequently if it's only ever updated once per second?

**Reply by wu-sheng** [reply]
> My question is why would it be called that frequently if it's only ever updated once per second?

This is how avg works. The total value divides reported time.

t1, value v1
t2, value v2
t3, value v3

The combine is doing (v1+v2+v3)/3(count of t).

**Reply by wu-sheng** [reply]
> My question is why would it be called that frequently if it's only ever updated once per second?

This is how avg works. The total value divides reported time.

t1, value v1
t2, value v2
t3, value v3

The combine is doing (v1+v2+v3)/3(count of t).

**Reply by BFergerson** [reply]
I get that. I'm trying to understand why `count` would reach `1707` if `JVMService` only ever reports `InstanceJvmCpuMetrics` once per second and I'm only running one JVM.

My expectation is that I would have a `count` of `60`, a `double_summation` of `60`, and a `double_value` of `1`.
Instead, I have a `count` of `1707`, a `double_summation` of `1707`, and a `double_value` of `1`.

I understand `double_summation`/`count` = `double_summary`. I just can't understand why `count` is `1707`. It should be `60`.

Would you not expect `count` to be `60`? If not, why is that?

**Reply by wu-sheng** [reply]
Is the time bucket at the minute level? Or day or hour level?

**Reply by BFergerson** [reply]
Ignoring the top two rows, they are all minute-level rows.

**Reply by wu-sheng** [reply]
Then it is new to me. Didn't face that.

**Reply by BFergerson** [reply]
I'll work on a reproducible example. I thought it was strange.

**Reply by wu-sheng** [reply]
Yes, this should not happen.

**Reply by BFergerson** [reply]
Sorry, turns out this is an issue with my fork. SkyWalking works perfectly by itself. Thanks for the help.
